modules/compiler/utils/include/compiler/utils/barrier_regions.h

```diff
@@ -173,9 +176,13 @@ class Barrier {
     LiveValuesHelper(Barrier const &b, llvm::BasicBlock *bb, llvm::Value *s)
         : barrier(b), gepBuilder(bb), barrier_struct(s) {}
 
-    /// @brief get a GEP instruction pointing to the given value in the barrier
-    /// struct.
-    llvm::Value *getGEP(const llvm::Value *live);
+    /// @brief get a GEP instruction pointing to the given value/idx pair in
+    /// the barrier struct.
+    llvm::Value *getGEP(const llvm::Value *live, unsigned member_idx = 0);
+
+    /// @brief Try and get a GEP instruction pointing to the given literal
```

Comment:
When it says Try - does it return nullptr if it cannot? Is that an expected possibility or an error?

Reply:
Returning `nullptr` is a possibility, as in `getGEP`, which this function just steals the `return nullptr` case from.

As for what it means for `getGEP` to return `nullptr`, I'm not sure. The APIs in this code aren't very well defined as they're all intended to be used internally. I think in practice `getGEP` is only used on variables that are contained within the live variables struct: variables contained within `live_vars_defs_in_kernel`, which are those contained within `whole_live_variables_set_`, which is also iterated over to insert the members into the structs. It's all a bit circuitous for my liking: both separate in terms of data and tracking, but also each are inextricably linked.